cpp/src/arrow/record_batch.cc

```diff
@@ -198,14 +198,20 @@ Result<std::shared_ptr<RecordBatch>> RecordBatch::MakeEmpty(
 }
 
 Result<std::shared_ptr<RecordBatch>> RecordBatch::FromStructArray(
-    const std::shared_ptr<Array>& array) {
+    const std::shared_ptr<Array>& array, MemoryPool* memory_pool) {
```

Comment:
It  might be a good idea to review the callsites to this function and ensure they pass the context's memory pool instead of just passing the default one automatically.

Reply:
Good catch.  Fortunately it seems we don't use this method too much.  I forwarded the parameter to two call sites in the JSON reader (although I think we will always be zero-copy in those cases it doesn't hurt).  There is one more call site in Datum::MakeExecBatch but it was already missing a memory pool (and implicitly using the default in other calls, e.g. cast) so I just left it as is.  I'm also not sure this code path is used anywhere.